modules/shared_data/src/main/scala/org/elpaca_metagraph/shared_data/Utils.scala

```diff
@@ -100,9 +100,16 @@ object Utils {
         password.value.value.toCharArray
       )
 
-  def randomString(length: Int): String = {
+  def randomString(input: String, length: Int): String = {
+    val hash = MessageDigest.getInstance("SHA-256")
+      .digest(input.getBytes("UTF-8"))
+      .map("%02x".format(_)).mkString
+
+    val seed = BigInt(hash.take(16), 16).toLong
+    val random = new Random(seed)
```

Comment:
I don't really know the requirements for the change, but do we really need calculate the hash for a seed? Seems a little overkill and is still deterministic on the input (unless that's what we look for).

Reply:
Yes, I believe the issue is that you need to seed the pseudo-random generator with a deterministic string so that all nodes will agree on the new value that is used for random string. Otherwise the nodes will fork.

Reply:
Yes, just like James described

Reply:
Have you considered using cats-effect Random? https://typelevel.org/cats-effect/docs/std/random
The Random that you use is side-effectful and not referentially transparent. (hence, not thread-safe by default and difficult to test)

Reply:
@marcinwadon I've updated to this Random, please take a look